### PR TITLE
fix issue with advi when running on GPU

### PR DESCRIPTION
**Pull request by huynhnguyen**
Hi folks,
It seems to me that advi fails when running on GPU due to the wrong format test value. Here is my fixing. It will be great if my code can help somebody.

### CODE REVIEW

File: pymc3/variational/advi.py [hunk 1]
```diff
@@ -182,7 +182,7 @@ def _calc_elbo(vars, model, n_mcsamples, random_seed):
 
     uw = tt.vector('uw')
     uw.tag.test_value = np.concatenate([inarray.tag.test_value,
-                                        inarray.tag.test_value])
+                                        inarray.tag.test_value]).astype(np.float32)
```
Comment (ferrine) [inline pymc3/variational/advi.py]: Why not `theano.config.floatX`

Comment (huynhnguyen) [reply pymc3/variational/advi.py]: It also works, I think.

Comment (ferrine) [inline pymc3/variational/advi.py]: It does, `floatX` is a string

Comment (huynhnguyen) [inline pymc3/variational/advi.py]: I prefer to use numpy because I plan to port pymc3 to Tensorflow in future.

Comment (twiecki) [reply pymc3/variational/advi.py]: Yes, that's better.

File: pymc3/variational/advi.py [hunk 2]
```diff
@@ -182,7 +182,7 @@ def _calc_elbo(vars, model, n_mcsamples, random_seed):
 
     uw = tt.vector('uw')
```
Comment (twiecki) [inline pymc3/variational/advi.py:183]: Can't the type be forced here already, perhaps by `dvector`? Or won't the test_value be casted correctly?

Comment (huynhnguyen) [reply pymc3/variational/advi.py:183]: I already try to cast <code>uw</code> but no fixing. It just a bit issue with init test_value default type.

File: pymc3/variational/advi.py (hunk 1 above)
Comment (huynhnguyen) [reply pymc3/variational/advi.py]: So it will up to you for fixing it. Do I have to change it and re-folk?

Comment (taku-y) [reply pymc3/variational/advi.py]: Could you change it and commit again?

Comment (huynhnguyen) [reply pymc3/variational/advi.py]: sure, no issue

Comment (twiecki) [reply pymc3/variational/advi.py]: @huynhnguyen We can merge once this is fixed.

Comment (huynhnguyen) [reply pymc3/variational/advi.py]: okay, just changed the type. Do not have machine to test at home but 95% confident it works.

**Comment by twiecki** [comment]
Thanks @huynhnguyen!

**Comment by huynhnguyen** [comment]
you are welcome

**Pull request merged**